### ISSUE TITLE
Precompiled binaries

### ISSUE THREAD

**Issue by Bluscream** [landave/TSIdentityTool#1]
Could you provide binaries for the most common OSes? (win,arm x86/64)?

**Comment by landave** [comment]
Why?
Since the tool has only a single dependency, it should be very easy to compile it yourself, so I don't see the need for precompiled binaries.

**Comment by Bluscream** [comment]
Not everyone has experience with unix compilers like gcc and a unix os. I managed to compile TeamspeakHasher but this seems to be a bit too much :/

**Comment by landave** [comment]
If you tell me where exactly you are stuck, I may be able to help you.

**Comment by Bluscream** [comment]
Can i compile it via the windows embedded shell using ubuntu or debian from the mircrosoft appstore and if not can i compile it via a ubuntu VM via VirtualBox?

**Comment by Bluscream** [comment]
Also im not even sure if i can compile a windows binary on linux :/

**Comment by landave** [comment]
> Can i compile it via the windows embedded shell using ubuntu or debian from the mircrosoft appstore and if not can i compile it via a ubuntu VM via VirtualBox?

If you are on Windows, you can use the Windows Subsystem for Linux (WSL), for example with Ubuntu. Alternatively, you can use a virtual machine (e.g. with VirtualBox) to compile, just as you suggested.

> Also im not even sure if i can compile a windows binary on linux :/

The compiled binary will **not** run natively on Windows, but this doesn't matter. Just compile TSIdentityTool on Linux (for example with WSL), and run it once to read out the public key of your TeamSpeak identity.

**Comment by Bluscream** [comment]
What do you think about a cross platform teamspeak plugin for it. You can access the stored identities via QSqliteManager from the databases Protobuf object

**Comment by landave** [comment]
It is probably not worth the effort. However, if you think that people might be interested in something like this, you are welcome to fork the project and write the plugin yourself.

**Comment by Bluscream** [comment]
Im going to check out your source indepth soon. Even tho i have some issues with my compiled version of TeamspeakHasher. Gonna create issues on the respective repo :)

**Comment by keyliin0** [comment]
Can anyone provide binaries i have problems compiling 
![image](https://user-images.githubusercontent.com/35576004/42134809-3dc0eb24-7d3a-11e8-9efc-cc438f37e0bf.png)

**Comment by landave** [comment]
It looks like you are missing the libtomcrypt/libtommath libraries. Try this:
``apt-get install libtomcrypt-dev libtommath-dev``

**Comment by keyliin0** [comment]
I tried but still getting the same issue , I'm using ubuntu on a virtualbox tho

**Comment by landave** [comment]
Could you paste the command you use to call the compiler?

**Comment by keyliin0** [comment]
gcc TSIdentityTool.c -o TSIdentityTool -l tommath -l tomcrypt

**Comment by landave** [comment]
Okay, that seems to be good. What is the very _first_ error that gcc prints?

**Comment by keyliin0** [comment]
![image](https://user-images.githubusercontent.com/35576004/42136141-3097f96a-7d4e-11e8-8f83-02adcc779500.png)
I'm not familiar with Linux but I guess it might be the version
edit : sorry wrong image,updated it

**Comment by landave** [comment]
What version of Ubuntu are you using? What is the output of the following:
``dpkg -s libtomcrypt-dev | grep Version``

**Comment by keyliin0** [comment]
Version: 1.17-7
I'll try to install another version , thank you

**Comment by landave** [comment]
Mhh, 1.17 is not the newest version, but it should be recent enough. Could you take a look at `/usr/include/tomcrypt_pk.h` and check whether this file contains a declaration of `ltc_ecc_is_valid_idx(int n)`?

**Comment by keyliin0** [comment]
Yes
Here is the file if you want to check it yourself https://pastebin.com/JcG9w0ne

**Comment by landave** [comment]
Mh, that looks good. Maybe the dev package does not depend on the library itself, so you missed it? Try:
``apt-get install libtomcrypt1 libtommath1``

**Comment by keyliin0** [comment]
It's all good now I installed another version and everything works fine now
Thank you

**Comment by landave** [comment]
So which version where you using before, and which is working now?

**Comment by keyliin0** [comment]
I was using Version: 1.17-7 and now I got the latest version of kali linux. i tried the commands you gave me and it works. Probably i messed it up when installing libtomcrypt cause what I did is I downloaded the libtomcrypt from github and run the command `make install`. Actually when i tried to use the last command you gave me on ubuntu it couldn't find the package so.. What i did now is just install gcc , libtomcrypt1 and libtommath1 then compile
It seems is just me not familiar with linux
Thank you for the help
